### DISCUSSION TITLE
Bind arrays in Sql Objects queries

### DISCUSSION THREAD

**Discussion by fedxyz** [jdbi/jdbi#2454] [Q&A] [answered]
Hi,

I am using jdbi with Postgres database and Postgres plugin installed. 
It's not clear to me to how to bind arrays in queries in Sql Objects and which data type are supported by default.

Have I to use `@Bind`?

Enum are supported ?
For example if I have an Enum `Test`:

```
@SqlQuery("select 'FOOBAR' = any(:p)")
Boolean tesMethod(@Bind("p") Test[] testParameter);
```

is valid/correct?  From my tests it seems to work but I don't find in the documentation that enum are supported by array types.

Another doubt, can I use for example `Long[]`, like:

```
@SqlQuery("select 1 = any(:p)")
Boolean tesMethod(@Bind("p") Long[] testParameter);
```

or I have to use `long[]` ?  

Thanks for the help

**Comment by stevenschlansker** [comment, marked as answer]
`@Bind` is used to change the name of the parameter binding. If the method parameter names are present (non-default compiler configuration), then `@Bind` is optional. See: https://jdbi.org/#compiling_with_parameter_names

With the Postgres plugin installed, any scalar type that the server supports arrays of should be supported. Enums (`Test[]`), primitives (`long[]`), and boxed (`Long[]`)  should all work.

When writing DAO code, I would encourage you to write test cases that cover your usage of Jdbi. For example, a unit test for your DAO interface would check that enums can be read and written, and that `Long`s are stored. That checks that you are invoking Jdbi in a way that you get the results you expect.
